superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndMetricSelect.tsx

```diff
@@ -104,6 +108,12 @@ const columnsContainAllMetrics = (
   );
 };
 
+export type DndMetricSelectProps = DndControlProps<ValueType> & {
+  savedMetrics: savedMetricType[];
+  columns: ColumnMeta[];
+  datasourceType?: DatasourceType;
+};
+
 export const DndMetricSelect = (props: any) => {
```

Comment:
TODO: get rid of this `any`. Currently `DndMetricSelectProps` will conflict with too many things within this function. This file needs a bigger overhaul.